include/bamit/IntervalNode.hpp

```diff
@@ -438,10 +442,12 @@ inline auto get_overlap_records(seqan3::sam_file_input<traits_type, fields_type,
     // Get the file position of the first record matching start query.
     get_overlap_file_position(input, node_list, start, end, file_position);
 
-    // Store reads which start before the end of the query, filtering out unmapped reads.
+    // Store reads which start before the end of the query, filtering out unmapped reads and reads within the interval which
+    // end before the start. Example: Read 1 goes from 100 - 200, Read 2 goes from 101 - 151. Both in the same node (median 150), but
+    // when searching for interval 160 - 200, Read 2 will be included in results.
```

Comment:
Isn't your description describing the bug, and the following is what you want to say?:
```suggestion
    // when searching for interval 160 - 200, Read 2 will not be included in results, as it is outside the query range.
```